### PR TITLE
Add unit tests for instance start call.

### PR DESCRIPTION
**Pull request by mikalstill**

### CODE REVIEW

File: shakenfist/external_api/app.py [hunk 1]
```diff
@@ -635,10 +635,14 @@ def post(self, name=None, cpus=None, memory=None, network=None,
...
 
+        # Only system can specify a uuid
+        if instance_uuid and get_jwt_identity() != 'system':
+            return error(401, 'Only system can specify an instance uuid')
+
         # If accessing a foreign namespace, we need to be an admin
         if get_jwt_identity() not in [namespace, 'system']:
             return error(401,
-                         'only admins can create resources in a different namespace')
+                         'Only admins can create resources in a different namespace')
```
Comment (mandoonandy) [inline shakenfist/external_api/app.py:645]: Currently errors start with lower case. Are we changing to upper case?

File: shakenfist/tests/test_external_api.py [hunk 2]
```diff
@@ -71,7 +76,7 @@ def test_post_auth_bad_parameter(self):
         self.assertEqual(400, resp.status_code)
         self.assertEqual(
             {
-                'error': "post() got an unexpected keyword argument 'keyyy'",
+                'error': 'post() got an unexpected keyword argument \'keyyy\'',
```
Comment (mandoonandy) [inline shakenfist/tests/test_external_api.py:79]: This is not very PEP8?

File: shakenfist/external_api/app.py (hunk 1 above)
Comment (mikalstill) [reply shakenfist/external_api/app.py:645]: Fixed.

File: shakenfist/tests/test_external_api.py (hunk 2 above)
Comment (mikalstill) [reply shakenfist/tests/test_external_api.py:79]: Fixed.

**Comment by mikalstill** [comment]
Comments addressed, please re-review.

File: shakenfist/tests/test_external_api.py [hunk 3]
```diff
@@ -71,7 +76,7 @@ def test_post_auth_bad_parameter(self):
         self.assertEqual(400, resp.status_code)
         self.assertEqual(
             {
-                'error': "post() got an unexpected keyword argument 'keyyy'",
+                'error': "post() got an unexpected keyword argument \'keyyy\'",
```
Comment (mandoonandy) [inline shakenfist/tests/test_external_api.py:79]: The slashes can be removed now

File: shakenfist/tests/test_external_api.py [hunk 4]
```diff
@@ -143,7 +148,7 @@ def test_auth_header_wrong(self):
                                 }))
         self.assertEqual(
             {
-                'error': "Bad Authorization header. Expected value 'Bearer <JWT>'",
+                'error': 'Bad Authorization header. Expected value \'Bearer <JWT>\'',
```
Comment (mandoonandy) [inline shakenfist/tests/test_external_api.py:151]: To be PEP8'erised and de-slashed

File: shakenfist/tests/test_external_api.py (hunk 3 above)
Comment (mikalstill) [reply shakenfist/tests/test_external_api.py:79]: Fixed.

File: shakenfist/tests/test_external_api.py (hunk 4 above)
Comment (mikalstill) [reply shakenfist/tests/test_external_api.py:151]: Fixed.

**Pull request merged**